### PR TITLE
Emergency pause improvements

### PR DESCRIPTION
**Pull request by 0xDEnYO**
# Which Jira task belongs to this PR?

# Why did I implement it this way?

# Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] This pull request is as small as possible and only tackles one problem
- [ ] I have added tests that cover the functionality / test the bug
- [ ] I have updated any required documentation

# Checklist for reviewer (DO NOT DEPLOY and contracts BEFORE CHECKING THIS!!!)

- [ ] I have checked that any arbitrary calls to external contracts are validated and or restricted
- [ ] I have checked that any privileged calls (i.e. storage modifications) are validated and or restricted
- [ ] I have ensured that any new contracts have had AT A MINIMUM 1 preliminary audit conducted on <date> by <company/auditor>

### CODE REVIEW

File: script/tasks/diamondEMERGENCYPause.sh [hunk 1]
```diff
@@ -151,6 +152,20 @@ function handleNetwork() {
   echoDebug "BLACKLIST=$BLACKLIST"
   echo ""
 
+  # check if the diamond is already paused by calling owner() function and analyzing the response
+  local RESPONSE=$(cast call "$DIAMOND_ADDRESS" "owner()" --rpc-url "$RPC_URL")
+  if [[ "$ACTION" != "unpause the diamond" && "$RESPONSE" == *"$DIAMOND_IS_PAUSED_SELECTOR"* || "$RESPONSE" == *"DiamondIsPaused"* ]]; then
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/tasks/diamondEMERGENCYPause.sh:174]: updated.

File: script/utils/diamondEMERGENCYPauseGitHub.sh [hunk 2]
```diff
@@ -65,6 +61,24 @@ function handleNetwork() {
     echo "[network: $NETWORK] diamond address found in deploy log file: $DIAMOND_ADDRESS"
   fi
 
+  # check if the diamond is already paused by calling owner() function and analyzing the response
+  local RESPONSE=$(cast call "$DIAMOND_ADDRESS" "owner()" --rpc-url "$RPC_URL")
+  if [[ "$RESPONSE" == *"$DIAMOND_IS_PAUSED_SELECTOR"* || "$RESPONSE" == *"DiamondIsPaused"* ]]; then
+      success "[network: $NETWORK] The diamond is already paused."
+      echo "<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< end network $NETWORK <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<"
+      exit 0
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/utils/diamondEMERGENCYPauseGitHub.sh:70]: updated

File: script/utils/diamondEMERGENCYPauseGitHub.sh [hunk 3]
```diff
@@ -116,6 +130,37 @@ function handleNetwork() {
   fi
 }
 
+function printStatus() {
+  local NETWORK="$1"
+
+  # get RPC URL for given network
+  local RPC_KEY="ETH_NODE_URI_$(tr '[:lower:]' '[:upper:]' <<<"$NETWORK")"
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/utils/diamondEMERGENCYPauseGitHub.sh:141]: told you to not report this issue again. Why are you not doing it?

File: .github/workflows/diamondEmergencyPause.yml [hunk 4]
```diff
@@ -1,12 +1,12 @@
 name: Pause PRODUCTION diamond (CAREFUL)
 
 on:
+  push:
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .github/workflows/diamondEmergencyPause.yml]: outdated

File: .github/workflows/diamondEmergencyPause.yml [hunk 5]
```diff
@@ -98,10 +96,10 @@ jobs:
...
-            :warning: 'ATTENTION - the emergency diamond pause action was just executed by ${{ github.actor }}'
-        env:
-          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK_DEV_SMARTCONTRACTS }}
+      # - name: Send Discord message
+      #   uses: Ilshidur/action-discord@0.3.2
+      #   with:
+      #     args: |
+      #       :warning: 'ATTENTION - the emergency diamond pause action was just executed by ${{ github.actor }}'
+      #   env:
+      #     DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK_DEV_SMARTCONTRACTS }}
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .github/workflows/diamondEmergencyPause.yml]: outdated

File: script/tasks/diamondEMERGENCYPause.sh [hunk 6]
```diff
@@ -134,6 +134,7 @@ function handleNetwork() {
   RPC_URL=$(getRPCUrl "$NETWORK")
 
   DIAMOND_ADDRESS=$(getContractAddressFromDeploymentLogs "$NETWORK" "production" "$DIAMOND_CONTRACT_NAME")
+  DIAMOND_ADDRESS="0xD3b2b0aC0AFdd0d166a495f5E9fca4eCc715a782"  # TODO: remove <<<<<<<<<---------------------------------------------------------------------------------------- (STAGING DIAMOND ON POL, ARB, OPT)
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/tasks/diamondEMERGENCYPause.sh]: outdated

File: script/utils/diamondEMERGENCYPauseGitHub.sh [hunk 7]
```diff
@@ -116,6 +130,37 @@ function handleNetwork() {
...
+
+  # get diamond address for this network
+  # DIAMOND_ADDRESS=$(getContractAddressFromDeploymentLogs "$NETWORK" "production" "LiFiDiamond")
+  DIAMOND_ADDRESS="0xD3b2b0aC0AFdd0d166a495f5E9fca4eCc715a782"  # TODO: remove <<<<<<<<<---------------------------------------------------------------------------------------- (STAGING DIAMOND ON POL, ARB, OPT)
+
+  # check if the diamond is paused by calling owner() function and analyzing the response
+  local RESPONSE=$(cast call "$DIAMOND_ADDRESS" "owner()" --rpc-url "$RPC_URL")
+  if [[ "$RESPONSE" == *"$DIAMOND_IS_PAUSED_SELECTOR"* || "$RESPONSE" == *"DiamondIsPaused"* ]]; then
+      success "[network: $NETWORK] The diamond is paused."
+      exit 0
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/utils/diamondEMERGENCYPauseGitHub.sh]: outdated

File: script/utils/diamondEMERGENCYPauseGitHub.sh [hunk 8]
```diff
@@ -65,6 +61,24 @@ function handleNetwork() {
     echo "[network: $NETWORK] diamond address found in deploy log file: $DIAMOND_ADDRESS"
   fi
 
+  # check if the diamond is already paused by calling owner() function and analyzing the response
+  local RESPONSE=$(cast call "$DIAMOND_ADDRESS" "owner()" --rpc-url "$RPC_URL")
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/utils/diamondEMERGENCYPauseGitHub.sh]: outdated

File: deployments/optimism.diamond.staging.json [hunk 9]
```diff
@@ -1,6 +1,10 @@
 {
   "LiFiDiamond": {
     "Facets": {
+      "0x17Bb203F42d8e404ac7E8dB6ff972B7E8473850b": {
+        "Name": "EmergencyPauseFacet",
+        "Version": "1.0.0"
+      },
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply deployments/optimism.diamond.staging.json:7]: i think your verification process is wrong. The facet is correctly deployed and has these functions.

File: script/utils/diamondEMERGENCYPauseGitHub.sh [hunk 10]
```diff
@@ -65,6 +58,32 @@ function handleNetwork() {
     echo "[network: $NETWORK] diamond address found in deploy log file: $DIAMOND_ADDRESS"
   fi
 
+  # check if the diamond is already paused by calling owner() function and analyzing the response
+  local RESPONSE=$(cast call "$DIAMOND_ADDRESS" "owner()" --rpc-url "$RPC_URL" 2>&1)
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply script/utils/diamondEMERGENCYPauseGitHub.sh:61]: please do not report this issue again for bash scripts (Separate variable declaration and assignment)

File: .github/workflows/diamondEmergencyPause.yml [hunk 11]
```diff
@@ -15,11 +15,11 @@ jobs:
...
-      - name: Check input
-        if: ${{ inputs.Warning != 'UNDERSTOOD' }}
-        run: |
-          echo -e "\033[31mYou must type 'UNDERSTOOD' to proceed.\033[0m"
-          exit 1
+      # - name: Check input
+      #   if: ${{ inputs.Warning == 'UNDERSTOOD' }}
+      #   run: |
+      #     echo -e "\033[31mYou must type 'UNDERSTOOD' to proceed.\033[0m"
+      #     exit 1
```
[the remark this answers is not in the document]
Comment (0xDEnYO) [reply .github/workflows/diamondEmergencyPause.yml]: outdated

**Review by ezynda3 — APPROVED** [review]

**Pull request merged**